src/repository.rs

```diff
@@ -591,4 +606,23 @@ mod test {
         read.read_to_end(&mut buf).expect("read target");
         assert_eq!(buf.as_slice(), data);
     }
+
+    #[test]
+    #[should_panic]
```

Comment:
I also wonder if this could use a `panic::catch_unwind` just on the final step to ensure we only catch the correct panic.